src/storage/file_manager.c

```diff
@@ -5613,6 +5664,304 @@ file_get_sticky_first_page (THREAD_ENTRY * thread_p, const VFID * vfid, VPID * v
   return NO_ERROR;
 }
 
+/*
+ * file_set_tde_algorithm () - set encryption algorithm in file header for TDE.
+ *
+ * return        : NO_ERROR, or ER_code
+ * thread_p (in)  : Thread entry
+ * vfid (in)      : File identifier
+ * tde_algo (in) : encryption algorithm - NONE, AES, ARIA
+ *
+ */
+int
+file_set_tde_algorithm (THREAD_ENTRY * thread_p, const VFID * vfid, TDE_ALGORITHM tde_algo)
+{
+  VPID vpid_fhead;
+  PAGE_PTR page_fhead = NULL;
+  FILE_HEADER *fhead = NULL;
+  int error_code = NO_ERROR;
+
+  if (!tde_Cipher.is_loaded && tde_algo != TDE_ALGORITHM_NONE)
```

Comment:
Could you give me an example? Do you mean `!(tde_Cipher.is_loaded || tde_algo == TDE_ALGORITHM_NONE)`?

Reply:
I think I misunderstood if it is not the error case in case of tde_Cipher.is_loaded == false && tde_algo == TDE_ALGORITHM_NONE. this condition is OK, right?

Reply:
Yes, if tde_algo is NONE, it is ok even if tde_cipher is not loaded.